lib/fdo.c

```diff
@@ -1099,30 +1099,19 @@ static bool _STATE_TO1(void)
 			if (rv->bypass && *rv->bypass == true) {
 				rvbypass = true;
 				break;
-			}
-			if (rv->owner_only && *rv->owner_only == true) {
+			} else if (rv->owner_only && *rv->owner_only == true) {
 				rvowner_only = true;
 				break;
-			}
-
-			if (rv->ip) {
+			} else if (rv->ip) {
 				ip = rv->ip;
-				rv = rv->next;
-				continue;
-			}
-			if (rv->dn) {
+			} else if (rv->dn) {
 				dns = rv->dn;
-				rv = rv->next;
-				continue;
-			}
-			if (rv->po) {
+			} else if (rv->po) {
 				port = *rv->po;
-				rv = rv->next;
-				continue;
-			}
-			if (rv->pr && (*rv->pr == RVPROTHTTPS || *rv->pr == RVPROTTLS)) {
+			} else if (rv->pr && (*rv->pr == RVPROTHTTPS || *rv->pr == RVPROTTLS)) {
```

Comment:
As per spec section 3.7 (Table ‑. RendezvousInfo Variables), the default behavior of RVProtocol is TLS. So we should set tls to false only if we receive RVProtocol as RVPROTHTTP, otherwise keep TLS enabled.